common/src/main/java/io/druid/timeline/LogicalSegment.java

```diff
@@ -19,8 +19,10 @@
 
 package io.druid.timeline;
 
+import io.druid.guice.annotations.PublicApi;
 import org.joda.time.Interval;
 
+@PublicApi
```

Comment:
It's supposed to return the same objects that it was given (but possibly filtered). So it doesn't need to be an extension point (people won't subclass it), just a public api (people may need to call methods on it).